zookeeper-contrib/zookeeper-contrib-loggraph/src/main/java/org/apache/zookeeper/graph/Log4JSource.java

```diff
@@ -115,20 +115,16 @@ public Log4JSourceIterator(Log4JSource src, long starttime, long endtime, Filter
 	}
 	
 	synchronized public long size() throws IOException {
-	    if (LOG.isTraceEnabled()) {
```

Comment:
I am not sure this is good.

Usually this pattern prevents a waste of resources in invocation of the heavier method: pass parameters on the stack, performing the call, boxing primitives.

Although slf4j can be good in early existing I doubt that the impact is negligible, especially in hot paths

I would prefer to ensure to have a consistent usage of this pattern (if logenabled...log) instead of removing it

Reply:
So in this particular case, it's am excellent candidate to be removed.  The only difference here is that the static string "size() called" is pushed onto the stack before the call.  That is immeasurably small.  In return we get less code that is less error prone.  There may also be some performance benefit from having a smaller code base when it comes to execution caching, so it may be faster with less code, though it's simply conjecture on my part.

"performing the call".  It has to perform a call no matter what.  It's either calling `isTraceEnabled()` or `trace()`.  So that is a wash.

There is no auto-boxing in this case as it is passing a static string.

Reply:
@eolivelli OK. So in this case you would like to see....

```
if (LOG.isTraceEnabled()) {
   LOG.trace("size() called");
}
```

Even though they are the exact same thing, though the second one pushes on argument onto the stack?  Do I understand that correctly?